studiopress/genesis-custom-blocks
# Fix an issue with the default value of 'Checkbox' and 'Toggle' fields

#### Background
* I introduced a bug into the newly released editor where the default of the Toggle and Checkbox fields is truthy instead of falsey
* The data type of the default was also wrong, though it was wrong before (also my fault 😂 ). It should be a `boolean`, but it was stored as `0` or `1` 

#### Changes
* Makes the default for Toggle and Checkbox fields `false`, how it used to be in GCB `1.0.3`: <img width="771" alt="here-false" src="https://user-images.githubusercontent.com/4063887/106514758-49c01b80-649a-11eb-8333-120fb6cc9b1a.png">

* Changes the data type of the default value from a string like `'0'` or `'1'` to a `boolean`
* But most of this diff is deleting dead code from the previous editor that I forgot to delete. It's slightly related to this fix, as fields have `->sanitize` and `->validate` functions that aren't used anymore and confusing.
* To test with a previously-created field, you might have to go to: new editor  > Toggle field > Default > check and uncheck, then click 'Update: 
![testing-up](https://user-images.githubusercontent.com/4063887/106416743-c1009b80-6417-11eb-8fa9-dbcade63ea42.gif)

#### Testing instructions
1. `composer install && npm install && npm run dev`
2. `/wp-admin` > Custom Blocks > Add New
3. Name the block 'Test Checkbox'
4. Add a field
5. Give it any 'Field Label'
6. Change its type to 'Checkbox': <img width="794" alt="expected-unchecked" src="https://user-images.githubusercontent.com/4063887/106515206-f69a9880-649a-11eb-90c4-d0b0dc05c58d.png">
7. Publish the block
8. Create a new post
9. Add the 'Test Checkbox' block you created: 
<img width="913" alt="unchecked-here" src="https://user-images.githubusercontent.com/4063887/106515353-2f3a7200-649b-11eb-866b-a234bba81740.png">
10. Repeat steps 2-8, except with a 'Toggle' field type 
11. Expected: it's untoggled by default: 
<img width="835" alt="untoggled-by-default" src="https://user-images.githubusercontent.com/4063887/106515594-87717400-649b-11eb-9d11-3966740346e5.png">
